Bottom: 1d798fba9499b29413c741c6fceb102e92cf3804
Top:    629e7fe00b602e5320ffad97b615d51e952e1922
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-02-19 15:30:41 +0000

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index 88afa1b..2bbff2a 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -309,6 +309,16 @@ original_poly (PolygonType * p)
       if (n == p->PointN - 1 ||
           (hole < p->HoleIndexN && n == p->HoleIndex[hole] - 1))
         {
+          if (contour == NULL)
+            {
+              printf ("How did that escape - did the loop iterate zero times??\n");
+              POLYGONPOINT_LOOP (p);
+                {
+                  printf ("Hello\n");
+                }
+              END_LOOP;
+              return NULL;
+            }
           poly_PreContour (contour, TRUE);
 
           /* make sure it is a positive contour (outer) or negative (hole) */
